src/views/tables/FillsTable.tsx

```diff
@@ -55,12 +55,53 @@ export enum FillsTableColumnKey {
   AmountTag = 'Amount-Tag',
   Total = 'Total',
   Fee = 'Fee',
+  ClosedPnl = 'ClosedPnl',
 
   // Tablet Only
   TypeAmount = 'Type-Amount',
   PriceFee = 'Price-Fee',
 }
 
+const calculateClosedPnl = (fill: FillTableRow) => {
+  const fee = parseFloat(fill.fee ?? '0');
+
+  // Old fills are not supported so we show -- instead of 0
+  if (fill.positionSizeBefore === undefined) {
+    return '--';
+  }
+
+  // No position before = opening trade, only fees realize
+  if (fill.positionSizeBefore === 0) {
+    return -fee;
+  }
+
+  // Check if position is reducing (opposite side)
+  const isReducing =
+    (fill.positionSideBefore === 'LONG' && fill.side === 'SELL') ||
+    (fill.positionSideBefore === 'SHORT' && fill.side === 'BUY');
```

Comment:
find the appropriate enum instead of checking against the string.

`IndexerPositionSide` and `IndexerOrderSide`